tests/integration/golang/helpers/s3.go

```diff
@@ -31,3 +35,71 @@ func NewS3Client(endpoint string) (*s3.Client, error) {
 		o.UsePathStyle = true
 	}), nil
 }
+
+// CreateBuckets creates the test bucekts.
+func CreateBuckets(s3Client *s3.Client) error {
+	for _, bucket := range testBuckets {
+		_, err := s3Client.CreateBucket(context.Background(), &s3.CreateBucketInput{
+			Bucket: aws.String(bucket),
+		})
+		if err != nil {
+			return eris.Wrapf(err, "failed to create bucket '%s'", bucket)
+		}
+	}
+	return nil
+}
+
+// RemoveBuckets removes the test buckets.
+func RemoveBuckets(s3Client *s3.Client) error {
+	for _, bucket := range testBuckets {
+		if err := removeBucket(s3Client, bucket); err != nil {
+			return eris.Wrapf(err, "failed to remove bucket '%s'", bucket)
+		}
+	}
+	return nil
+}
+
+// removeBucket removes a bucket and its objects.
+func removeBucket(s3Client *s3.Client, bucket string) error {
+	// Delete all objects in the bucket
+	var objectIDs []types.ObjectIdentifier
+	paginator := s3.NewListObjectsV2Paginator(s3Client, &s3.ListObjectsV2Input{
+		Bucket: aws.String(bucket),
+	})
+	for paginator.HasMorePages() {
+		page, err := paginator.NextPage(context.Background())
+		if err != nil {
+			return eris.Wrapf(err, "error paging objects in bucket '%s'", bucket)
+		}
+		for _, object := range page.Contents {
+			objectIDs = append(objectIDs, types.ObjectIdentifier{Key: object.Key})
+		}
+	}
+	if len(objectIDs) > 0 {
+		_, err := s3Client.DeleteObjects(context.Background(), &s3.DeleteObjectsInput{
+			Bucket: aws.String(bucket),
+			Delete: &types.Delete{Objects: objectIDs},
+		})
+		if err != nil {
+			return eris.Wrapf(err, "failed to delete objects in bucket '%s'", bucket)
+		}
+	}
+
+	// Delete the bucket
+	if _, err := s3Client.DeleteBucket(context.Background(), &s3.DeleteBucketInput{
+		Bucket: aws.String(bucket),
+	}); err != nil {
+		return eris.Wrapf(err, "failed to delete bucket '%s'", bucket)
+	}
+	waiter := s3.NewBucketNotExistsWaiter(s3Client)
+	if err := waiter.Wait(
+		context.Background(),
+		&s3.HeadBucketInput{
+			Bucket: aws.String(bucket),
+		},
+		time.Second,
```

Comment:
We should wait for longer than a second here, to avoid tests failing because of some disk/cpu/memory performance issue on a GitHub Actions hosted runner.
I'd suggest 10 seconds to be on the safe side, as in most cases this will be close to instant.

Reply:
will do